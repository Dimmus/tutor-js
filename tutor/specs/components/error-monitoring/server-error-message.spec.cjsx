{React, SnapShot} = require '../helpers/component-testing'

# jest.mock('../../../src/helpers/router')
# TutorRouter = require '../../../src/helpers/router'

# {CourseActions, CourseStore} = require '../../../src/flux/course'

# TestRouter = require '../helpers/test-router'
# COURSE_ID = '1'
# COURSE = require '../../../api/user/courses/1.json'

ErrorMessage = require '../../../src/components/error-monitoring/server-error-message'

# Wrapper = (props) ->
#   <span>{props.body}</span>

describe 'Error monitoring: server-error message', ->

  beforeEach ->
    # CourseActions.loaded(COURSE, COURSE_ID)
    # @course = CourseStore.get(COURSE_ID)
    # TutorRouter.currentParams.mockReturnValue({courseId: COURSE_ID})
    # TutorRouter.makePathname.mockReturnValue('/go/to/dash')
    @props =
      status: 404
      statusMessage: "Not Found"
      config:
        method: 'none'
        url: 'non-url'
        data: 'code: Red'

  it 'renders for errors with status 500', ->
    @props.status = 500
    wrapper = shallow(<ErrorMessage {...@props} />)
    expect(wrapper.text()).to.include('500')
    undefined

  it 'matches snapshot', ->
    component = SnapShot.create(<ErrorMessage {...@props} />)
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
