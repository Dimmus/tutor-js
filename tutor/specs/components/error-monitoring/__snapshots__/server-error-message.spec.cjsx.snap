exports[`Error monitoring: server-error message matches snapshot 1`] = `
<div
  className="server-error">
  <h3>
    An error with code 
    404
     has occured
  </h3>
  <p>
    Please visit 
    <a
      href="http://openstax.force.com/support?l=en_US&c=Products%3ATutor&cu=1&fs=ContactUs&q=Hello!%0AI%20ran%20into%20a%20problem%20on%0ANode.js%20(darwin%3B%20U%3B%20rv%3Av7.2.1)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20at%20about%3Ablank.%0A%0AHere%20is%20some%20additional%20info%3A%0A404%20with%20Not%20Found%20for%20none%20on%20non-url%20with%0Acode%3A%20Red."
      target="_blank">
      our support page
    </a>
     to file a bug report.
  </p>
  <p>
    Additional error messages returned from the server is:
  </p>
  <pre
    className="response">
    Not Found
  </pre>
  <div
    className="request">
    <kbd>
      none
    </kbd>
     on 
    non-url
     
    <span>
      with 
      <pre>
        code: Red
      </pre>
    </span>
  </div>
</div>
`;
